diff --git a/include/rfl/yaml/Reader.hpp b/include/rfl/yaml/Reader.hpp
index e6c2cfd..1715f39 100644
--- a/include/rfl/yaml/Reader.hpp
+++ b/include/rfl/yaml/Reader.hpp
@@ -66,7 +66,7 @@ struct Reader {
   }
 
   bool is_empty(const InputVarType& _var) const noexcept {
-    return !_var.node_ && true;
+    return (!_var.node_ || _var.node_.Type() == YAML::NodeType::Null) && true;
   }
 
   template <class T>
